crates/noirc_frontend/src/hir/def_collector/dc_crate.rs

```diff
@@ -333,11 +334,15 @@ fn type_check_global_consts(
     for (file_id, stmt_id) in global_const_ids {
         let mut type_check_errs = Vec::new();
         let _stmt_type = type_check(interner, &stmt_id, &mut type_check_errs);
-        let type_check_err_diagnostics: Vec<_> = type_check_errs.clone().into_iter()
-                                        .map(|error| error.into_diagnostic(interner)).collect();
-    
+        let type_check_err_diagnostics: Vec<_> = type_check_errs
+            .clone()
+            .into_iter()
+            .map(|error| error.into_diagnostic(interner))
+            .collect();
```

Comment:
If you're collecting into a Vec you can use `vecmap` instead of `.into_iter().map(...).collect()`, also do these need to be cloned?